spec/gilded_rose_spec.js

```diff
@@ -1,7 +1,267 @@
+'use strict'
+
+function resetItems(){
+  items = []
+}
+
+function populateItems(){
+  items.push(new Item('+5 Dexterity Vest', 10, 20));
+  items.push(new Item('Aged Brie', 2, 0));
+  items.push(new Item('Elixir of the Mongoose', 5, 7));
+  items.push(new Item('Sulfuras, Hand of Ragnaros', 0, 80));
+  items.push(new Item('Backstage passes to a TAFKAL80ETC concert', 15, 20));
+  items.push(new Item('Conjured Mana Cake', 3, 20));
+}
+
 describe("Gilded Rose", function() {
+  describe("items have a sell-in value", () => {
+    it("should initialize a quality", function() {
+      resetItems()
+      populateItems()
+
+      expect(items[0].sell_in).toEqual(10)
+      expect(items[1].sell_in).toEqual(2)
+      })
+    })
+  describe("items have a quality value", () => {
+    it("should initialize a quality", () => {
+      resetItems()
+      populateItems()
+
+      expect(items[4].quality).toEqual(20)
+      expect(items[5].quality).toEqual(20)
+    })
+  })
+  describe("Dexterity Vest", () => {
+    it("sell_in value decrease by 1 each day", () => {
+      resetItems()
+      populateItems()
+      update_quality()
+
+      expect(items[0].sell_in).toEqual(9)
+      })
+
+    it("quality value decrease by 1 each day", () => {
+      resetItems()
+      populateItems()
+      update_quality()
+
+      expect(items[0].quality).toEqual(19)
+      })
+    it("sell_in is negative after 11 days", () => {
+      resetItems()
+      populateItems()
+      let time = items[0].sell_in
+      for (var i = 0; i < time+1; i++){
```

Comment:
Avoid for loops. 
Reading this literally I would think that sell_in *always* goes negative after 11 days. Is that really true?

From reading this, how do I know:

- The number of times this will loop?
- *why* it is looping
- *how* the looping is important (hint: it isn't!)
- *why* 11 days has anything to do with what is being tested (hint: it doesn't)